harness/determined/experimental/pytorch_lightning/adapter.py

```diff
@@ -0,0 +1,79 @@
+import sys
+from inspect import signature
+from typing import Any, Callable, Dict, Sequence, Union
+
+import pytorch_lightning as ptl
+import torch
+
+from determined.pytorch import PyTorchTrial, PyTorchTrialContext
+
+TorchData = Union[Dict[str, torch.Tensor], Sequence[torch.Tensor], torch.Tensor]
+HyperparamsProvider = Callable[[str], Any]
+
+
+def bail(msg: str = "", fail: bool = True):
```

Comment:
blocking: please create an except class for invalid trial in `harness/determined/error.py` and just use `raise` directly since it is what is done across the codebase.